arch/arc/core/reset.S

```diff
@@ -86,6 +86,24 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_and__start,__start)
 	kflag r0
 #endif
 
+#if defined (CONFIG_CPU_HS4X) && defined (CONFIG_ARC_HAS_DSP)
+	/* case of HS4XD cpu */
+	lr	 r0, [_ARC_V2_LPB_BUILD]
+	breq r0, 0, skip_lpb_setup
+	mov  r0, 1	/* disable LPB for HS4XD */
+	sr   r0, [_ARC_V2_LPB_CTRL]
+skip_lpb_setup:
+#endif
```

Comment:
Could you please check if there is simple way to move it to SoC code? Preferably with just using ASM macro.

Another thing - we need to move this after cache maitanance code in reset.S